intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/problems/ProblemLevelRemappingManager.kt

```diff
@@ -0,0 +1,99 @@
+import com.fasterxml.jackson.databind.DatabindException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.jetbrains.plugin.structure.base.problems.PluginProblem.Level.ERROR
+import com.jetbrains.plugin.structure.base.problems.PluginProblem.Level.WARNING
+import java.io.IOException
+import java.net.URL
+import kotlin.reflect.KClass
+
+interface ProblemLevelRemappingManager {
+  fun initialize(): LevelRemappingDefinitions
+}
+
+fun levelRemappingFromClassPathJson(): JsonUrlProblemLevelRemappingManager {
+  val pluginProblemsJsonUrl = JsonUrlProblemLevelRemappingManager::class.java.getResource(PLUGIN_PROBLEMS_FILE_NAME)
+    ?: throw IOException("Plugin problem level remapping definition cannot be found at <$PLUGIN_PROBLEMS_FILE_NAME>")
+  return JsonUrlProblemLevelRemappingManager(pluginProblemsJsonUrl)
+}
+
+private const val INTELLIJ_PROBLEMS_PACKAGE_NAME = "com.jetbrains.plugin.structure.intellij.problems"
+const val PLUGIN_PROBLEMS_FILE_NAME = "plugin-problems.json"
+
+class JsonUrlProblemLevelRemappingManager(private val pluginProblemsJsonUrl: URL) : ProblemLevelRemappingManager {
+  private val json = ObjectMapper()
+
+  @Throws(IOException::class)
+  override fun initialize() = load()
+
+  @Throws(IOException::class)
+  fun load(): LevelRemappingDefinitions {
+    val definitions = LevelRemappingDefinitions()
+    try {
+      val rawRemapping: Map<String, Map<String, String>> = json.readValue(pluginProblemsJsonUrl)
+
+      rawRemapping.map { (problemSetName: String, problemRemapping: Map<String, String>) ->
+        val levelRemapping = problemRemapping.mapNotNull { (problemId, problemLevel) ->
+          val pluginProblemKClass = resolveClass(problemId) ?: return@mapNotNull null
+          when (problemLevel) {
+            "ignore" -> pluginProblemKClass to IgnoredLevel
+            "warning" -> pluginProblemKClass to StandardLevel(WARNING)
+            "error" -> pluginProblemKClass to StandardLevel(ERROR)
+            else -> null
+          }
+        }.toMap()
+        definitions[problemSetName] = levelRemapping
+      }
+    } catch (e: IOException) {
+      throw IOException("Cannot load plugin problems definitions from <$pluginProblemsJsonUrl>", e)
+    } catch (e: StreamReadException) {
+      throw IOException("Cannot parse plugin problems definitions from JSON in <$pluginProblemsJsonUrl>", e)
+    } catch (e: DatabindException) {
+      throw IOException("Cannot deserialize plugin problems definitions from JSON in <$pluginProblemsJsonUrl>", e)
+    }
+    return definitions
+  }
+
+
+  /**
+   * Resolves the problem ID to a fully qualified class name.
+   *
+   * Example: `ForbiddenPluginIdPrefix` to `com.jetbrains.plugin.structure.intellij.problems.ForbiddenPluginIdPrefix`
+   */
+  private fun resolveClass(problemId: String): KClass<out Any>? {
+    val fqn = "$INTELLIJ_PROBLEMS_PACKAGE_NAME.$problemId"
```

Comment:
Added `com.jetbrains.plugin.structure.base.problems`. Note that there are additional package names (such as `rider`).,Let's add that later, since there might be duplicate names in the different packages.

Reply:
why not use FQN everywhere?

Reply:
Using FQN everywhere. 

Updated the JSON.